homeassistant/components/binary_sensor/august.py

```diff
@@ -58,14 +72,69 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
     data = hass.data[DATA_AUGUST]
     devices = []
 
+    from august.lock import LockDoorStatus
+    for door in data.locks:
+        for sensor_type in SENSOR_TYPES_DOOR:
+            state_provider = SENSOR_TYPES_DOOR[sensor_type][2]
+            if state_provider(data, door) is LockDoorStatus.UNKNOWN:
+                _LOGGER.debug(
+                    "Not adding sensor class %s for lock %s ",
+                    SENSOR_TYPES_DOOR[sensor_type][1], door.device_name
+                )
+                continue
+
+            _LOGGER.debug(
+                "Adding sensor class %s for %s",
+                SENSOR_TYPES_DOOR[sensor_type][1], door.device_name
+            )
+            devices.append(AugustDoorBinarySensor(data, sensor_type, door))
+
     for doorbell in data.doorbells:
-        for sensor_type in SENSOR_TYPES:
-            devices.append(AugustBinarySensor(data, sensor_type, doorbell))
+        for sensor_type in SENSOR_TYPES_DOORBELL:
+            _LOGGER.debug("Adding doorbell sensor class %s for %s",
+                          SENSOR_TYPES_DOORBELL[sensor_type][1],
+                          doorbell.device_name)
+            devices.append(
+                AugustDoorbellBinarySensor(data, sensor_type,
+                                           doorbell)
+            )
 
     add_entities(devices, True)
 
 
-class AugustBinarySensor(BinarySensorDevice):
+class AugustDoorBinarySensor(BinarySensorDevice):
+    """Representation of an August Door binary sensor."""
+
+    def __init__(self, data, sensor_type, door):
+        """Initialize the sensor."""
+        self._data = data
+        self._sensor_type = sensor_type
+        self._door = door
+        self._state = None
+
+    @property
+    def is_on(self):
+        """Return true if the binary sensor is on."""
+        return self._state
+
+    @property
+    def device_class(self):
+        """Return the class of this device, from component DEVICE_CLASSES."""
+        return SENSOR_TYPES_DOOR[self._sensor_type][1]
+
+    @property
+    def name(self):
+        """Return the name of the binary sensor."""
+        return "{} {}".format(self._door.device_name,
+                              SENSOR_TYPES_DOOR[self._sensor_type][0])
+
+    def update(self):
+        """Get the latest state of the sensor."""
+        state_provider = SENSOR_TYPES_DOOR[self._sensor_type][2]
+        self._state = state_provider(self._data, self._door)
```

Comment:
This does not return a boolean, however it's returned directly from is_on. That is wrong.

Reply:
As I have written twice now, and this is the third time, you need to return `False` from the available property if the status is unknown.

Reply:
@balloob,

My sincere apologies, but I thought that this coding was doing what you were asking for returning False if the status is unknown.

For
```
state_provider = SENSOR_TYPES_DOOR[self._sensor_type][2]
```
sets state_provider to method _retrieve_door_state.

Thus
`self._state = state_provider(self._data, self._door)`
for the door sensor would be:
`self._state = _retrieve_door_state(data, lock)`

Method _retrieve_door_state(data, lock) returns:
`return doorstate == LockDoorStatus.OPEN`

Thus self._state would be set to True (if door is open) or False (if door state is anything else).

Is this not what you mean? I looked at some other code and for example in spc.py I see for is_on the return is:
`return self._zone.input == ZoneInput.OPEN`

Thus I think that the current coding sets the state indeed then to True or False with only setting it to True if the returned status is Open, otherwise returning False. 

If this is incorrect then my sincere apologies and I would ask if you could point me to some coding of a sensor that does it correctly. I looked at a few and they all return True or False bu doing it directly in is_on or by setting self._state somewhere else to True or False and then just returning self._state

Thank you.

Reply:
@balloob, thanks for pointing me in the right direction for the available property. It is defined now for the doorsensor and set to False if self._state is unknown otherwise it is set to True.